### PR TITLE
Fix codecov job failure

### PR DESCRIPTION
**Pull request by hklai**
wait was failing apparently when the sub process had exited, which in
turn caused the job to fail, even if no test had actually failed.

The fix is to write the err to a tmp file in the parallel sub-processes
and later use them to report falures.

### CODE REVIEW

File: bin/codecov.sh [hunk 1]
```diff
@@ -43,22 +44,19 @@ mkdir -p "$COVERAGEDIR"
...
     -coverprofile="${COVERAGEDIR}/${filename}.cov" \
     -covermode=atomic "${1}" \
-    | tee "${COVERAGEDIR}/${filename}.report" ) &
-  local pid=$!
-  PKGS[${pid}]=${1}
-  PIDS+=("${pid}")
+    | tee "${COVERAGEDIR}/${filename}.report"  && RC=$? || RC=$?
+
+  if [[ ${RC} != 0 ]]; then
+    echo "${1}" | tee "${COVERAGEDIR}/${filename}.err"
```
Comment (ostromart) [inline bin/codecov.sh]: Should these be cleaned up? Otherwise if you rerun you'll see fails from previous runs.

Comment (hklai) [reply bin/codecov.sh]: The output and err files are in the randomly generated $COVERAGEDIR in /tmp.

**Pull request merged**